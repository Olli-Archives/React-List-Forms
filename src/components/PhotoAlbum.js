import React from 'react';
import PropTypes from 'prop-types';
import Photos from './Photos';
import Header from './Header';

export default function PhotoAlbum({ album }){
  const { title, photos } = album;
  console.log('title passed to header', title);

  return (
    <ul>
      <Header title={title}/>
      <Photos photos={photos}/>
    </ul>
  );
}

PhotoAlbum.propTypes = {
  album: PropTypes.object.isRequired
}
;
