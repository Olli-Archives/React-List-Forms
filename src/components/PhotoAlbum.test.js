import PhotoAlbum from './PhotoAlbum';
import { shallow } from 'enzyme';
import React from 'react';

const photosArray = ['https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg', 'https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg', 'https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg'];
const title = 'Best Album Ever';
const album = { 
  title:title,
  photos: photosArray
};



describe('PhotoAlbum component', ()=>{
  it('creates an album witha title and an array of images', ()=>{
    const wrapper = shallow(<PhotoAlbum album={album}/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
