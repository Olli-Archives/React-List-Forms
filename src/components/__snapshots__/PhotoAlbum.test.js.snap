// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoAlbum component creates an album witha title and an array of images 1`] = `
<ul>
  <Header
    title="{title}"
  />
  <Photos
    photos={
      Array [
        "https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg",
        "https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg",
        "https://ichef.bbci.co.uk/news/936/cpsprodpb/E4CB/production/_95717585_happy.jpg",
      ]
    }
  />
</ul>
`;
